src/Skia/Perspex.Skia/FormattedTextImpl.cs

```diff
@@ -75,36 +108,206 @@ public Size Measure()
+                        nextChar = _text[curOff + measured];
+
+                    if (nextChar != ' ')
+                    {
+                        //Perform scan for the last space and end the line there
+                        for (int si = curOff + measured - 1; si > curOff; si--)
+                        {
+                            if (_text[si] == ' ')
+                            {
+                                measured = si - curOff;
+                                extraSkip = 1;
+                                //Rects[si] = SkRect();
+                                break;
+                            }
+                        }
+                    }
+
+
+                }
+
+                PerspexFormattedTextLine line = new PerspexFormattedTextLine();
+                line.Start = curOff;
+                line.Length = measured;
+                line.Width = lineWidth;
+                line.Height = lineHeight;
+                line.Top = curY;
+
+                if (line.Width < 0)
+                    line.Width = _skiaRects[line.Start + line.Length - 1].Right;
+
+                //Build character rects
+                for (int i = line.Start; i < line.Start + line.Length; i++)
+                {
+                    float prevRight = 0;
+                    if (i != line.Start)
+                        prevRight = _skiaRects[i - 1].Right;
+
+                    subString = _text.Substring(line.Start, i - line.Start + 1);
+                    float w = Paint.MeasureText(subString);
+
+                    SKRect rc;
+                    rc.Left = prevRight;
+                    rc.Right = w;
+                    rc.Top = line.Top;
+                    rc.Bottom = line.Top + line.Height;
+                    _skiaRects[i] = rc;
+                }
+
+                subString = _text.Substring(line.Start, line.Length);
+                line.Width = Paint.MeasureText(subString);
+
+                _skiaLines.Add(line);
+
+                curY += lineHeight; // + mLeading;
+                curOff += measured + extraSkip;
+            }
+
+            // Now convert to Perspex data formats
             _lines.Clear();
             _rects.Clear();
             float maxX = 0;
-            for (var c = 0; c < _shared->LineCount; c++)
+
+            for (var c = 0; c < _skiaLines.Count; c++)
             {
-                var w = _shared->Lines[c].Width;
+                var w = _skiaLines[c].Width;
                 if (maxX < w)
                     maxX = w;
-                _lines.Add(new FormattedTextLine(_shared->Lines[c].Length, _shared->Lines[c].Height));
+
+                _lines.Add(new FormattedTextLine(_skiaLines[c].Length, _skiaLines[c].Height));
             }
+
             for (var c = 0; c < _text.Length; c++)
-                _rects.Add(_shared->Bounds[c].ToRect());
-            if (_shared->LineCount == 0)
+                _rects.Add(_skiaRects[c].ToPerspexRect());
+
+            if (_skiaLines.Count == 0)
```

Comment:
Make braces consistent. From [.net core style guidelines](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md):

A single line statement block can go without braces but the block must be properly indented on its own line and it must not be nested in other statement blocks that use braces (See issue 381 for examples).